thunder/torch/__init__.py

```diff
@@ -259,6 +260,12 @@ def to(
         if dtype is not None:
             dtype = to_dtype(dtype)
             a = prims.convert_element_type(a, dtype)
+        if memory_format is not None:
+            # NOTE not sure if we need to handle torch.preserve_format explicitly
```

Comment:
What does copying to a different device (like CUDA -> CPU) do when the original tensor is noncontiguous?

Reply:
> when the original tensor is noncontiguous?

I think the protocol here is that, if a copy is happening, it's always going to be packed tensor at the destination.
The original tensor only suggests a layout and whether or not it packed isn't important.

Reply:
> What does copying to a different device (like CUDA -> CPU) 

The behavior should also be the same whether it's moving across device or on the same device with `copy == True`.

Reply:
```
>>> y = x.to(copy=True, memory_format=torch.preserve_format)
>>> y.stride()
(36, 12, 3, 1)
>>> x.stride()
(60, 20, 5, 2)
```